admin/common_auth/views.py

```diff
@@ -21,6 +24,47 @@ def login(request):
     context = {'form': form}
     return render(request, 'login.html', context)
 
+
 def logout(request):
     logout_user(request)
     return redirect('auth:login')
+
+
+# Permission restriction based on staff (only staff can register new users). Use is_staff boolean?
+def register(request):
```

Comment:
Since we're already returning from multiple places in this function, let's restructure the function to avoid such deep nesting, using a "gating" pattern:

```python
def register(request):
    if not request.user.is_staff:
        return redirect('auth:login')

    if not request.method != 'POST':
        return render(request, 'register.html', context)

    form = CustomUserRegistrationForm(request.POST)
    if not form.is_valid():
        return render(request, 'register.html', context, status=400)

    pass # process form
    return redirect('auth:register')
```